### PR TITLE
.gitmodules: add opentelemetry-cpp as a submodule 

### PR DESCRIPTION
**Pull request by ideepika**
- removes compile time fetch for opentelemetry
- updates to use opentelemetry remote instead of unofficial patched remote

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

with the PR's merged in opentelemetry we do not require to maintain our own
remote:
https://github.com/open-telemetry/opentelemetry-cpp/pull/1100/
https://github.com/open-telemetry/opentelemetry-cpp/pull/1020

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -72,3 +72,6 @@
 [submodule "src/utf8proc"]
 	path = src/utf8proc
 	url = https://github.com/JuliaStrings/utf8proc
+[submodule "src/jaegertracing/opentelemetry-cpp"]
```
Comment (tchaikov) [inline .gitmodules:75]: the commit message is not accurate. it includes the items not covered by this change.

File: cmake/modules/BuildOpentelemetry.cmake [hunk 2]
```diff
@@ -6,7 +6,7 @@ function(target_create _target _lib)
 endfunction()
 
 function(build_opentelemetry)
-  set(opentelemetry_SOURCE_DIR "${PROJECT_SOURCE_DIR}/src/opentelemetry-cpp")
+  set(opentelemetry_SOURCE_DIR "${PROJECT_SOURCE_DIR}/src/jaegertracing/opentelemetry-cpp")
```
Comment (tchaikov) [inline cmake/modules/BuildOpentelemetry.cmake:9]: could you please use `cmake: ` as the prefix of the title of the commit message? also, might want to explain why we'd want to use submodule in place of `git fetch`.

File: .gitmodules (hunk 1 above)
Comment (tchaikov) [reply .gitmodules:75]: also, could you please include `opentelemetry-cpp submodule` in the commit message as proposed by https://jenkins.ceph.com/job/ceph-pr-submodules/67495/console ?

**Review by adamemerson — APPROVED** [review]
Thank you!

**Review by tchaikov — APPROVED** [review]

**Comment by ideepika** [comment]
https://shaman.ceph.com/builds/ceph/wip-opentelemetry-submodule/0edeeeb67cec8f80365751a657baa14322ce5de9/
failures unrelated.

**Pull request merged**